### ISSUE TITLE
Unexpected behavior with link_df and its memory argument when frames missing

### ISSUE THREAD

**Issue by kmdouglass** [soft-matter/trackpy#370]
Hi all,
I have found that the `memory` argument to `link_df()` is essentially ignored on a test case I made consisting of a single particle that disappears for a couple frames and then returns in **trackpy 0.3.0**. In the example below, the last appearance of the particle is erroneously grouped into the track for the first four particles, despite its appearance after a time longer than the `memory` argument.

```python
import pandas as pd
import trackpy as tp

test_df = pd.DataFrame({'x' : [10, 20, 0, 0, 20], 'y' : [10, 20, 20, 0, 0], 'frame' : [0, 1, 2, 3, 6]})

tp.link_df(test_df, 25, memory = 1)

assert test_df['particle'].max() >= 1, 'Error: all features grouped into one particle.'
```

An AssertionError will be raised because every element of the `test_df['particle']` column will be zero. However, if I add another particle such that the gap between frames 3 and 6 is filled, I get the expected results, where the last feature is not joined to the first four.

```python
import pandas as pd
import trackpy as tp

test_df = pd.DataFrame({'x' : [10, 20, 0, 100, 0, 110, 90, 90, 20, 110], 'y' : [10, 20, 20, 100, 0, 110, 110, 90, 0, 90], 'frame' : [0, 1, 2, 2, 3, 3, 4, 5, 6, 6]})

tp.link_df(test_df, 25, memory = 1)

assert test_df['particle'].max() == 2, 'Error: Unexpected number of one particles.'
```

The above will not raise the AssertionError.

**Comment by kmdouglass** [comment]
My apologies. I think this is the same as #292

**Comment by caspervdw** [comment]
Yes, this is indeed fixed in the next release, v0.3.1, which will come out very soon!
